### PR TITLE
snapshot: add CLI documentation

### PR DESCRIPTION
**Pull request by LegendreM**
add documentation for:

* `--snapshot-path`
* `--snapshot-interval-sec`
* `--init-from-snapshot`
* `--recover-from-snapshot`

### CODE REVIEW

File: guides/advanced_guides/configuration.md [hunk 1]
```diff
@@ -232,3 +237,51 @@ Depending on the OS, it is either the size that will be allocated on launch or t
 [To know more about storage in MeiliSearch look at this guide](/resources/about_storage.md)
 
 **Default value**: `107374182400` (100 GiB)
+
+### Schedule Snapshot Creation
+
+**CLI option**: `--snapshot-path`
+
+The directory path where meilisearch will create snapshot,
```
Comment (bidoubiwa) [inline guides/advanced_guides/configuration.md:245]:
```suggestion
The directory path where MeiliSearch will create snapshots.
```

File: guides/advanced_guides/configuration.md [hunk 2]
```diff
@@ -232,3 +237,51 @@ Depending on the OS, it is either the size that will be allocated on launch or t
...
+
+If this command is not provided, snapshoting is deactivated.
+
+### Snapshot interval
+
+**CLI option**: `--snapshot-interval-sec`
+
+defines time gap (seconds) between each snapshot creation.
```
Comment (bidoubiwa) [inline guides/advanced_guides/configuration.md:253]:
```suggestion
Defines the time gap (seconds) between each snapshot creation.
```

File: guides/advanced_guides/configuration.md [hunk 3]
```diff
@@ -232,3 +237,51 @@ Depending on the OS, it is either the size that will be allocated on launch or t
...
+
+Require `--snapshot-path` to be defined.
```
Comment (matboivin) [inline guides/advanced_guides/configuration.md:255]:
```suggestion
Requires `--snapshot-path` to be defined.
```

File: guides/advanced_guides/configuration.md (hunk 2 above)
Comment (matboivin) [reply guides/advanced_guides/configuration.md:253]: What about?

```suggestion
Defines the time gap in seconds between each snapshot creation.
```

File: guides/advanced_guides/configuration.md [hunk 4]
```diff
@@ -232,3 +237,51 @@ Depending on the OS, it is either the size that will be allocated on launch or t
...
+
+**Default value**: `86400` (1 day)
+
+### Init from snapshot
+
+**CLI option**: `--init-from-snapshot`
+
+The snapshot file path to import.
```
Comment (matboivin) [inline guides/advanced_guides/configuration.md:263]:
```suggestion
The path of the snapshot file to import.
```

File: guides/advanced_guides/configuration.md [hunk 5]
```diff
@@ -232,3 +237,51 @@ Depending on the OS, it is either the size that will be allocated on launch or t
...
+
+This command stop the process if:
```
Comment (matboivin) [inline guides/advanced_guides/configuration.md:265]:
```suggestion
This command will stop the process if:
```

File: guides/advanced_guides/configuration.md [hunk 6]
```diff
@@ -232,3 +237,51 @@ Depending on the OS, it is either the size that will be allocated on launch or t
...
+
+- a database already exist
```
Comment (matboivin) [inline guides/advanced_guides/configuration.md:267]:
```suggestion
- a database already exists
```

File: guides/advanced_guides/configuration.md [hunk 7]
```diff
@@ -232,3 +237,51 @@ Depending on the OS, it is either the size that will be allocated on launch or t
...
+- no snapshot exist at provided path.
```
Comment (matboivin) [inline guides/advanced_guides/configuration.md:268]:
```suggestion
- No snapshot exists in the given path.
```

File: guides/advanced_guides/configuration.md [hunk 8]
```diff
@@ -232,3 +237,51 @@ Depending on the OS, it is either the size that will be allocated on launch or t
...
+
+If this command is not provided, no snapshot will be imported.
```
Comment (matboivin) [inline guides/advanced_guides/configuration.md:270]:
```suggestion
If this command is not called, no snapshot will be imported.
```

File: guides/advanced_guides/configuration.md [hunk 9]
```diff
@@ -232,3 +237,51 @@ Depending on the OS, it is either the size that will be allocated on launch or t
...
 
 **Default value**: `107374182400` (100 GiB)
+
+### Schedule Snapshot Creation
+
+**CLI option**: `--snapshot-path`
+
+The directory path where meilisearch will create snapshot,
+
+If this command is not provided, snapshoting is deactivated.
```
Comment (matboivin) [inline guides/advanced_guides/configuration.md:247]:
```suggestion
If this command is not called, snapshoting is deactivated.
```

File: guides/advanced_guides/configuration.md [hunk 10]
```diff
@@ -232,3 +237,51 @@ Depending on the OS, it is either the size that will be allocated on launch or t
...
+
+If this command is not provided, no snapshot will be imported.
+
+Uncompatible with `--recover-from-snapshot`
+
+### Recover from snapshot
+
+**CLI option**: `--recover-from-snapshot`
+
+The snapshot file path to import.
```
Comment (matboivin) [inline guides/advanced_guides/configuration.md:278]:
```suggestion
The path of the snapshot file to import.
```

File: guides/advanced_guides/configuration.md [hunk 11]
```diff
@@ -232,3 +237,51 @@ Depending on the OS, it is either the size that will be allocated on launch or t
...
+
+This command skip importation and continue the process if:
```
Comment (matboivin) [inline guides/advanced_guides/configuration.md:280]:
```suggestion
This command will skip importing a snapshot and continue the process if:
```

File: guides/advanced_guides/configuration.md [hunk 12]
```diff
@@ -232,3 +237,51 @@ Depending on the OS, it is either the size that will be allocated on launch or t
...
+
+- a database already exist
```
Comment (matboivin) [inline guides/advanced_guides/configuration.md:282]:
```suggestion
- A database already exists
```

File: guides/advanced_guides/configuration.md [hunk 13]
```diff
@@ -232,3 +237,51 @@ Depending on the OS, it is either the size that will be allocated on launch or t
...
+- no snapshot exist at provided path.
```
Comment (matboivin) [inline guides/advanced_guides/configuration.md:283]:
```suggestion
- No snapshot exists in the given path.
```

File: guides/advanced_guides/configuration.md [hunk 14]
```diff
@@ -232,3 +237,51 @@ Depending on the OS, it is either the size that will be allocated on launch or t
...
+
+If this command is not provided, no snapshot will be imported.
```
Comment (matboivin) [inline guides/advanced_guides/configuration.md:285]:
```suggestion
If this command is not called, no snapshot will be imported.
```

**Comment by LegendreM** [comment]
@bidoubiwa @matboivin, I made the changes.
this documentation is linked to [this feature](https://github.com/meilisearch/MeiliSearch/pull/829) which will be stabilized in meilisearch v0.14.X

**Pull request merged**